pycognaize/common/numeric_parser.py

```diff
@@ -82,6 +82,8 @@ def parse_regular_float(self):
             # if it doesn't work use more complicated logic
             if self.removed_sign:
                 self.parsed = float(self.raw.strip()) * self.sign
+            if re.split(r'\D+', self.raw)[-1] > 2:
```

Comment:
Pre-compile the regex pattern to make it faster. Make the exception more explicit.